proj-sys/Cargo.toml

```diff
@@ -12,6 +12,8 @@ links = "proj"
 rust-version = "1.70"
 
 [dependencies]
+libsqlite3-sys = { version = "0.28", features = ["bundled"] }
```

Comment:
Based on your description, I was expecting this to be an optional dependency. What am I misunderstanding?

Reply:
It's unfortunately not possible to keep that as optional dependency as the build script can decide at runtime to build libproj from source. See [here](https://github.com/georust/proj/pull/190#issuecomment-1936465465) for more details.

Reply:
@michaelkirk Does this answer your question? Or would you like something to be changed here?

Reply:
Well ideally we wouldn't be compiling a thing that we don't need. I understand that it's complicated though.

Will `libsqlite3-sys` build sqlite3 even if it's already been installed on the system (e.g. via a package manager)?

Reply:
@michaelkirk Yes with the `bundled` feature flag enabled `libsqlite3-sys` will always build `libsqlite3` from source and statically link it. They do not provide a option that performs runtime detection whether the crate is installed or not and that falls back to compiling if the crate is not found. That means with the current version we will always build `libsqlite3` from source, but we will only link that version if we also build libproj from source. 

That written: I had another idea how we could maybe solve that problem in a better way. We could not enable the `bundled` feature flag in `proj-sys` at all. We can then use the `DEP_SQLITE3_*` variables to check at runtime whether or not the `libsqlite3-sys` crate build a libsqlite3 version from source by checking if these environment variables exist. That would allow users to control how libsqlite3 should be linked, as they then can control the `bundled` flag there on their own as feature flags are additive. If these variables are not set we would fall back to trying to get a system version of libsqlite3 for the build, which is the current behavior. If that sounds more acceptable I would change the PR to use that approach instead.

Reply:
@michaelkirk I've pushed https://github.com/georust/proj/pull/190/commits/44a503869709ea793a5f8452bfb9e2a99b7160fc which essentially implements what was described in my previous comment. Is there anything else required? I would like to move this forward to unblock the [gdal bundling PR](https://github.com/georust/gdal/pull/517)

Reply:
@weiznich Just so I'm clear: there's now no unneeded sqlite build unless the `bundled` feature is enabled?

Reply:
Yes, I changed it that way that it only uses the statically build libsqlite3 version if the user specified somewhere else in their dependency tree that libsqlite3 should be bundled.

Reply:
> uses the statically build

So it still builds sqlite every time, but discards the static build if it's not needed?

Reply:
No, it doesn't configure the bundled build for sqlite at all.

If we reach the stage where we build `libproj` from source is just checks if someone else has configured that `libsqlite3-sys` should build libsqlite3 from source and if that's the case it will use these build artifacts. That check is done by these two `DEP_*` variables, which are set by `libsqlite3-sys` for the bundled build. For that to work `proj-sys` needs to depend on `libsqlite3-sys` (otherwise we won't see the variables), but that shouldn't be a problem as that only runs the build script of that crate, which by default will only handle that libsqlite3 is linked to the binary, not that it is build from source.

Reply:
Great, thanks for clarifying!